std/range/package.d

```diff
@@ -2465,6 +2465,114 @@ Take!(Repeat!T) repeat(T)(T value, size_t n)
 }
 
 /**
+Given callable ($(XREF traits, isCallable)) $(D fun), create as a range
+whose front is defined by successive calls to $(D fun()).
+This is especially useful to call function with global side effects (random
+functions), or to create ranges expressed as a single delegate, rather than
+an entire $(D front)/$(D popFront)/$(D empty) structure.
+$(D fun) maybe be passed either a template alias parameter (existing
+function, delegate, struct type defining static $(D opCall)... ) or
+a run-time value argument (delegate, function object... ).
+The result range models an InputRange ($(XREF range, isInputRange).
+The resulting range will call $(D fun()) on every call to $(D front),
+and only when $(D front) is called, regardless of how the range is
+iterated. 
+It is advised to compose with either $(XREF algorithm,cache) or 
+$(XREF array,array), or to use in a (by-value) foreach loop.
```

Comment:
Not sure what `by-value` means here.
Also I think you need to compose __sth.__ with __...__.

```
It is advised to compose generate with either $(XREF algorithm,cache) or $(XREF array,array), or to use it in a foreach loop.
```